pubsub/api/PubsubTest/PubsubTest.cs

```diff
@@ -42,15 +42,60 @@ public class PubsubTest : IDisposable
             Command = "Pubsub"
         };
 
+        /// <summary>
+        /// Handle a special race condition that can occur when deleting
+        /// something:
+        /// 1. Delete request times out.
+        /// 2. Delete operation continues on server and succeeds.
+        /// 3. Later requests to delete the same entity see NotFound error.
+        /// </summary>
+        /// <param name="delete">The delete operation to run.</param>
+        /// <returns>An action to run inside Eventually().</returns>
+        Action HandleDeleteRace(Action delete)
+        {
+            bool sawTimeout = false;
+            return () =>
+            {
+                if (!sawTimeout)
+                {
+                    try
+                    {
+                        delete();
+                    }
+                    catch (Grpc.Core.RpcException e)
+                    when (e.Status.StatusCode == StatusCode.DeadlineExceeded)
+                    {
+                        sawTimeout = true;
+                        throw;
+                    }
+                }
+                else
+                {
+                    try
+                    {
+                        delete();
+                    }
+                    catch (Grpc.Core.RpcException e)
+                    when (e.Status.StatusCode == StatusCode.NotFound)
+                    {
+                        // Earlier timeout request that deleted the thing
+                        // actually succeeded on the server.
+                    }
+                }
+            };
+        }
+
         void IDisposable.Dispose()
         {
             foreach (string topicId in _tempTopicIds)
             {
-                Eventually(() => Run("deleteTopic", _projectId, topicId));
+                Eventually(HandleDeleteRace(() =>
+                    Run("deleteTopic", _projectId, topicId)));
             }
             foreach (string subscriptionId in _tempSubscriptionIds)
             {
-                Eventually(() => Run("deleteSubscription", _projectId, subscriptionId));
+                Eventually(HandleDeleteRace(() =>
+                    Run("deleteSubscription", _projectId, subscriptionId)));

```

Comment:
I think you should reverse the order of subscriptions deletion and topic deletion.
This is what I've found in documentation [here](https://cloud.google.com/pubsub/docs/admin#delete_a_topic):
**When you delete a topic, its subscriptions are not deleted, and the subscription's message backlog is available for subscribers. After a topic is deleted, its subscriptions have the topic name _deleted-topic_.**